src/components/utility/HeadingLinksCopyToClipboard/helpers.ts

```diff
@@ -0,0 +1,45 @@
+import { copyIcon } from './copyIconSVG'
+
+export const getHeadingsFromDocument = () => {
+	const mainArticleContainerSelector = '.main-section'
+	const headingsSelectors = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6']
+	headingsSelectors.forEach(
+		(heading, index) =>
+			(headingsSelectors[index] = mainArticleContainerSelector + ' ' + heading)
+	)
+	const headings = document.querySelectorAll<HTMLHeadingElement>(
+		headingsSelectors.join(', ')
+	)
+	return headings
+}
+
+const copyButtonClickHandler = (headingId: string) => {
+	try {
+		window.location.hash = headingId
+		navigator.clipboard.writeText(window.location.href)
```

Comment:
this needs "await" keyword in order to run it async

Reply:
but maybe I'm wrong

Reply:
This method returns a promise, so it won't hurt adding await there